core/data-plane/data-plane-util/src/main/java/org/eclipse/edc/connector/dataplane/util/sink/OutputStreamDataSinkFactory.java

```diff
@@ -37,12 +37,17 @@ public boolean canHandle(DataFlowRequest request) {
     }
 
     @Override
-    public @NotNull Result<Boolean> validate(DataFlowRequest request) {
-        return canHandle(request) ? Result.success(true) :
+    public @NotNull Result<Void> validate(DataFlowRequest request) {
+        return canHandle(request) ? Result.success() :
                 Result.failure(String.format("%s: Cannot handle destination data address with type: %s",
                         getClass().getSimpleName(), request.getDestinationDataAddress().getType()));
     }
 
+    @Override
+    public @NotNull Result<Void> validateRequest(DataFlowRequest request) {
+        return DataSinkFactory.super.validateRequest(request);
```

Comment:
the logic should go in this new method (`validateRequest`) and `validate` should call this, because in the future we'll remove `validate`. Same for all the other implementations